### PR TITLE
WIP: Mempool specification

### PR DESCRIPTION
**Pull request by ethanfrey**
Writing a public specification of the mempool module.

* Include overview and configuration options.

### CODE REVIEW

File: mempool/docs/config.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+[mempool]
+recheck_empty = false
+```
+
+
+## Recheck
+
+`--mempool.recheck=false` (default: true)
+
+`--mempool.recheck_empty=false` (default: true)
```
Comment (ethanfrey) [inline mempool/docs/config.md]: @ebuchman Do we want to make this false by default?

mempool.go:368 suggests it is used for debugging

Comment (ebuchman) [reply mempool/docs/config.md]: Maybe. But currently the mempool is supposed to be configured so a correct proposer will only propose valid txs. Apps that use EndBlock could invalidate transactions in the mempool without the mempool knowing, and so a correct proposer could return invalid txs.

We just need to be clear either way. There's def a performance benefit to setting it false (ie. we won't replay all our mempool txs when an empty block is committed ...)

File: mempool/docs/config.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Mempool Configuration
```
Comment (melekes) [inline mempool/docs/config.md]: there is https://github.com/tendermint/tendermint/pull/792. let's try to avoid duplication. thanks

Comment (melekes) [reply mempool/docs/config.md]: @zramsay maybe you can incorporate some things below into your PR

File: mempool/docs/messages.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+the actual 40-byte tx. Prefix of 350=0x015e byte tx is:
+`0x0102015e...` followed by the actual 350 byte tx.
+
+(Please see the [go-wire repo](https://github.com/tendermint/go-wire#an-interface-example) for more information)
+
+## RPC Messages
+
+Mempool exposes `CheckTx([]byte)` over the RPC interface.
+
+It can be posted via `broadcast_commit`, `broadcast_sync` or
```
Comment (melekes) [inline mempool/docs/messages.md]: these are not the same things... broadcast_commit calls both CheckTx and waits for DeliverTx result. not sure this should be here. i think this belongs to RPC API!

**Comment by ebuchman** [comment]
Merging, will clean up later

**Comment by zramsay** [comment]
do we want any of these rendered in RTD?

**Pull request merged**